### PR TITLE
IGAPP-1027: Save jpal tracking code

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

- [x] Check whether previously persisted code is deleted on native if clicking on https://integreat.app/jpal (without a tracking code)

### CODE REVIEW

File: web/src/utils/safeLocalStorage.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 let hasLocalStorageSupport: boolean | null = null
 
+export const I18N_LANGUAGE_KEY = 'i18nextLng'
+export const PAGE_FORCE_REFRESHED_KEY = 'pageForceRefreshed'
```
Comment (steffenkleinle) [inline web/src/utils/safeLocalStorage.ts:4]: was changed from `page-force-refresh`, should I keep the old one, delete the new one or just change it as it currently is? leads to two values saved in the local storage

File: web/src/routes/JpalTrackingPage.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, { ReactElement, useEffect } from 'react'
+import { useNavigate, useParams } from 'react-router-dom'
+
+import safeLocalStorage, { JPAL_TRACKING_CODE_KEY } from '../utils/safeLocalStorage'
+
+const JpalTrackingPage = (): ReactElement => {
```
Comment (f1sh1918) [inline web/src/routes/JpalTrackingPage.tsx:6]:
```suggestion
const JpalTrackingPage = (): null => {
```

File: web/src/routes/JpalTrackingPage.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    if (trackingCode) {
+      safeLocalStorage.setItem(JPAL_TRACKING_CODE_KEY, trackingCode)
+    } else {
+      safeLocalStorage.removeItem(JPAL_TRACKING_CODE_KEY)
+    }
+
+    navigate('/', { replace: true })
+  }, [trackingCode, navigate])
+
+  return <></>
```
Comment (f1sh1918) [inline web/src/routes/JpalTrackingPage.tsx:20]:
```suggestion
  return null
```

File: web/src/routes/JpalTrackingPage.tsx (hunk 2 above)
Comment (f1sh1918) [reply web/src/routes/JpalTrackingPage.tsx:6]: returning null is more performant than returning fragment

File: web/src/routes/JpalTrackingPage.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React, { ReactElement, useEffect } from 'react'
```
Comment (f1sh1918) [inline web/src/routes/JpalTrackingPage.tsx:1]:
```suggestion
import { useEffect } from 'react'
```

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
Works fine on chrome. Just some minor improvements. We do not need a release note?

**Comment by steffenkleinle** [comment]
> We do not need a release note?

Because imo this is only a technical task paving the way for another issue. Will write a release note for https://issues.tuerantuer.org/browse/IGAPP-1025 though.

**Review by f1sh1918 — APPROVED** [review]
works fine tested on chrome

**Review by SteffiStuffel — APPROVED** [review]
Tested on chrome

**Review by SteffiStuffel — APPROVED** [review]
Not results shown currently in the logs, therefore not possible to test. Code looks good.

**Comment by steffenkleinle** [comment]
>     * [x]  Check whether previously persisted code is deleted on native if clicking on https://integreat.app/jpal (without a tracking code)

Currently not working on native, created https://issues.tuerantuer.org/browse/IGAPP-1045

**Comment by steffenkleinle** [comment]
> Not results shown currently in the logs, therefore not possible to test. Code looks good.

Yes, not yet included in the requests, see #709. Also this is not yet implemented in the backend. However, #709 is testable as described here: https://github.com/digitalfabrik/integreat-app/pull/709#issuecomment-1173580607

**Pull request merged**